### PR TITLE
[MetricsAdvisor] Added tests for the AnomalyDetectionConfiguration CRUD operations

### LINKED ISSUE #15915: [MetricsAdvisor] Write tests for the AnomalyDetectionConfiguration CRUD operations
Add tests for the service methods and related models.
- CreateMetricAnomalyDetectionConfiguration
- GetMetricAnomalyDetectionConfiguration
- GetMetricAnomalyDetectionConfigurations
- UpdateMetricAnomalyDetectionConfiguration
- DeleteMetricAnomalyDetectionConfiguration

~~Also: add snapshot validation to the `MetricAnomalyFeedback` tests.~~ (moved to https://github.com/Azure/azure-sdk-for-net/issues/17585)

### PR DESCRIPTION
**Pull request by kinelski**
Fixes https://github.com/azure/azure-sdk-for-net/issues/15915.

Tests ended up very large because we're testing a complex model. It takes multiple lines for setup and some more for validation.

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyDetectionConfigurationTests.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+            Assert.That(() => adminClient.CreateDetectionConfiguration(null), Throws.InstanceOf<ArgumentNullException>());
+        }
+
+        [Test]
+        public void CreateDetectionConfigurationRespectsTheCancellationToken()
+        {
+            MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient();
+
+            var config = new AnomalyDetectionConfiguration(FakeGuid, "configName", new ());
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyDetectionConfigurationTests.cs:37]: just curious, is there any test that will verify the required input for `AnomalyDetectionConfiguration`?

Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyDetectionConfigurationTests.cs:37]: There isn't. However, we'll change some of our model's designs to have parameterless constructors (including `AnomalyDetectionConfiguration`), so these tests would be discarded. This is feedback we got during the archboard review (https://github.com/Azure/azure-sdk-for-net/issues/16321).

Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyDetectionConfigurationTests.cs:37]: We'll still have models that may need validation after that, though. That's a good point, so just opened https://github.com/Azure/azure-sdk-for-net/issues/17650 to track this. We can work on that after the design change.

**Review by maririos — APPROVED** [review]
LGTM. It will be good if someone with full understanding of the service (from other language?) will check as I am not super confident with all configurations

**Comment by kinelski** [comment]
> 
> 
> LGTM. It will be good if someone with full understanding of the service (from other language?) will check as I am not super confident with all configurations

Just added others as reviewers in case they want to contribute. I'll keep it open for more time.

**Review by kristapratico — APPROVED** [review]
looks good!

**Pull request merged**